runtime/examples/memtest/memtest.cpp

```diff
@@ -0,0 +1,177 @@
+#include <array>
+#include <iostream>
+#include <tapasco.hpp>
+#include <iomanip>
+#include <climits>
+
+using namespace tapasco;
+
+constexpr int random_time_ms = 1000;
+constexpr int batch_iterations = 1000;
+constexpr int latency_iterations = 100000;
+constexpr bool ecc_memory = true;
+
+constexpr int PE_ID = 321;
+constexpr int col_width = 20;
+constexpr char space = ' ';
+
+constexpr int batch_min_length = 10;
+constexpr int batch_max_length = 25;
+
+
+constexpr int random_byte_length_c = 10;
+constexpr int random_byte_length[random_byte_length_c] = {8, 16, 32, 64, 128, 256, 512, 1024, 2048, 4096};
+
+double calcSpeed(unsigned long amount, unsigned long cycles, float clock) {
+  return (amount / (cycles / (clock * 1000000.0))) / (1024*1024*1024);
+}
+
+void executeRandomBenchmark(tapasco::Tapasco &tapasco, int op, unsigned long cycles, int byte_length, float designclk) {
+  char text[20];
+  unsigned long ret = -1;
+  tapasco::RetVal<unsigned long> ret_val(ret);
+  auto job = tapasco.launch(PE_ID, ret_val, 3 + op, cycles, byte_length, 1234, 5678);
+  job();
+  unsigned long total_data = byte_length * ret;
+  double iops = ret / (cycles / (designclk * 1000000.0));
+  sprintf(text, "%.2fMIOPS", iops / 1000000);
+  std::cout << left << setw(col_width) << setfill(space) << text;
+  sprintf(text, "%#.2fGiB/s", calcSpeed(total_data, cycles, designclk));
+  std::cout << left << setw(col_width) << setfill(space) << text;
+  sleep(1);
+}
+
+void benchmarkRandom(tapasco::Tapasco &tapasco, float designclk, unsigned long cycles) {
+
+  std::cout << "Initializing memory (required for ECC memory)" << std::endl;
+  if (ecc_memory) {
+    auto job = tapasco.launch(PE_ID, 4, 2 * cycles, random_byte_length[random_byte_length_c - 1], 1234, 1234);
+    job();
+    sleep(1);
```

Comment:
What are the sleeps used for?

Reply:
If they are necessary it might be good to change to the C++ versions, e.g. `https://en.cppreference.com/w/cpp/thread/sleep_for`